Userland/Libraries/LibGfx/ImageFormats/JPEGLoader.cpp

```diff
@@ -102,18 +102,18 @@ using Marker = u16;
  */
 struct Macroblock {
     union {
-        i32 y[64] = { 0 };
-        i32 r[64];
+        i16 y[64] = { 0 };
```

Comment:
do you happen to know where the ITU doc says that these are signed?

Reply:
> A.1.2 Sample precision
A sample is an integer with precision P bits, with any value in the range 0 through 2^P – 1. All samples of all components within an image shall have the same precision P. Restrictions on the value of P depend on the mode of operation, as specified in B.2 to B.7.

Which means that the input data is unsigned.

> A.3.1 Level shift
Before a non-differential frame encoding process computes the FDCT for a block of source image samples, the samples shall be level shifted to a signed representation by subtracting 2(P – 1), where P is the precision parameter specified in B.2.2. Thus, when P = 8, the level shift is by 128; when P = 12, the level shift is by 2048.

This makes values be in $\mathbb {Z}$.

> B.2.2 Frame header syntax
P: Sample precision – Specifies the precision in bits for the samples of the components in the frame.